sqlglot/parser.py

```diff
@@ -2245,6 +2247,16 @@ def _parse_table_part() -> t.Optional[exp.Expression]:
             )
 
         table = _parse_table_part()
+
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#table_names
+        if self.TABLE_NAME_ALLOWS_DASHES and isinstance(table, exp.Identifier):
```

Comment:
why is this done here, shouldn't this be in parse_table_parts?

Reply:
Dashes are only allowed to the leftmost identifier according to the documentation, so I did this here to 1) capture that and 2) reduce unneeded overhead when parsing the rest of the parts.

Reply:
If this seems like unnecessary visual noise to have in the base parser, I can factor it out and have it only in BigQuery.

Reply:
let's just put this in bigquery because it's so unstandard

Reply:
done